### PR TITLE
Adding Addressabled LED class example

### PR DESCRIPTION
**Pull request by mrguisamuel**
Issue #49 requests some WPILib examples adapted from Java.
I am adding the Addressable LED class example here.
The code has been tested only in the simulator only, and the results were:
image

### CODE REVIEW

File: addressableled/robot.py
Comment (auscompgeek) [inline addressableled/robot.py:1]: We avoid using `from wpilib import ...` in our examples, as it's clearer to refer to things in the top level (such as `wpilib.run`) via the module name. Doing this also makes it easier to build on the examples.

File: addressableled/robot.py
Comment (auscompgeek) [inline addressableled/robot.py:9]: Our examples don't prefix member variables with `m_`, as the mandatory `self.` already implies it's a member variable.

File: addressableled/robot.py
Comment (auscompgeek) [inline addressableled/robot.py:38]: I think this will keep append to the `ledData` list whilst the robot code is running, forever using up more and more memory... until it eventually crashes.

File: addressableled/robot.py
Comment (auscompgeek) [inline addressableled/robot.py:28]: Why is this check necessary? This doesn't seem to exist in the Java or C++ examples.

**Review by auscompgeek — CHANGES_REQUESTED** [review]
Thanks for the PR with the screenshot! Do you have a recording of it too? I believe it should animate.

A few things need to be addressed before this can be merged in. You'll also need to run the autoformatter.

File: addressableled/robot.py
Comment (mrguisamuel) [reply addressableled/robot.py:1]: okay!

File: addressableled/robot.py
Comment (mrguisamuel) [reply addressableled/robot.py:9]: okay!

File: addressableled/robot.py
Comment (mrguisamuel) [reply addressableled/robot.py:28]: I am using this check because setData argument method requests an array with a length equal to or less than the buffer.
In Java examples, it is only requested without an array, so it won't be necessary to check before.
But this was initially the form that understood the Java example.

File: addressableled/robot.py
Comment (mrguisamuel) [reply addressableled/robot.py:38]: yeah
Should I run this function first at robotInit (rainbow)?
In this way, I could do an array of LEDData and it won't run every time.

File: addressableled/robot.py
Comment (mrguisamuel) [inline addressableled/robot.py:41]: Here, `self.m_ledData[i]` and `i`, are referenced using the `index` argument of `setHSV`.
[In Java, it appears.](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj/AddressableLEDBuffer.html)

File: addressableled/robot.py
Comment (mrguisamuel) [reply addressableled/robot.py:28]: By the way, I'm a beginner.

Comment (virtuald) [reply addressableled/robot.py:28]: In the C++ examples, they have a [fixed size array](https://github.com/wpilibsuite/allwpilib/blob/cb33bd71dfdfa9444c372e5a0141a3e6f298a4e5/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp#L17), but that'll be challenging to do in python.. and Java too, which is probably why they created the AddressableLEDBuffer class. 

I would initialize `ledData` like so:

```python
self.ledData = [AddressableLED.LEDData() for i in range(kLEDBuffer)]
```

Then later on you can just access `self.ledData[i]` instead of appending. Also removes the need for the length check.

Comment (mrguisamuel) [reply addressableled/robot.py:28]: Cool!

**Comment by mrguisamuel** [comment]
> Thanks for the PR with the screenshot! Do you have a recording of it too? I believe it should animate.

Here is the animation
Animation

File: addressableled/robot.py
Comment (auscompgeek) [inline addressableled/robot.py:35]: Thoughts on using the more Pythonic `enumerate`, @virtuald? Or would that perhaps be too confusing?

File: addressableled/robot.py
Comment (auscompgeek) [inline addressableled/robot.py:13]:
```suggestion
        self.ledData = [wpilib.AddressableLED.LEDData() for _ in range(kLEDBuffer)]
```

**Review by auscompgeek — APPROVED** [review]
LGTM! I have one comment I'd like a second opinion on, but I'm fine either way.

**Comment by TheTripleV** [comment]
Future idea: Can we have AddressableLED maintain its own internal storage to make it easier to use?

File: addressableled/robot.py
Comment (virtuald) [reply addressableled/robot.py:35]: This is fine.

**Comment by virtuald** [comment]
@TheTripleV That sounds like a bug for the wpilibc issue tracker? Java has an addressableledbuffer, but it doesn't really make things easier either.

File: addressableled/robot.py
Comment (OliverW10) [inline addressableled/robot.py:47]: should this be 180 instead of 100?

Comment (virtuald) [reply addressableled/robot.py:47]: Good catch!

**Comment by virtuald** [comment]
One last change needed for this: the test needs to be added to the `run_tests.sh` script.

File: addressableled/robot.py
Comment (mrguisamuel) [reply addressableled/robot.py]: thx @OliverW10
I didn't see this value :)

**Comment by mrguisamuel** [comment]
> One last change needed for this: the test needs to be added to the `run_tests.sh` script.

Here are the results
image
image

**Pull request merged**
